### ISSUE TITLE
Incompatible dependencies 

### ISSUE THREAD

**Issue by Samurai336** [gramps-project/gramps-webapi#421]
There is a dependency conflict when trying to install grapmps-webapi dependencies.
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
flake8 5.0.4 requires importlib-metadata<4.3,>=1.1.0; python_version < "3.8", but you have importlib-metadata 4.13.0 which is incompatible.
```

Did the following to get this issue on Linux Mint:

Once repository is cloned  

1. In grapmps-webapi project root setup a new virtual environment set to python 3.7 and activate environment.  I used pyenv with pyenv-virtualenv plugin

2. Edit your virtual environments pyvenv.cfg file (for me it was in ~/.pyenv/versions/<my-virtual-envs-name>/pyvenv.cfg) to set `include-system-site-packages` to `True` and save. 

3.  Install dev dependencies `pip install -r requirements-dev.txt` 

4.  Install gramps-webapi and dependencies `pip3 install -e .`   

This should create the conflict error listed above. Not using --user on the  editable dependency install because it in the virtual environment it would create a lack of sys.path precedence to gramps-webapi (see https://github.com/pypa/pip/issues/573)

**Comment by Samurai336** [comment]
python3.8 does not have this dependency conflict so does the project need to up its python dependency requirement?

**Comment by DavidMStraub** [comment]
Can you try updating flake8? It's just a development dependency without version pin...

**Comment by Samurai336** [comment]
it installs but I get an error warning me 
```
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
gramps-webapi 1.1.8 requires importlib-metadata==4.13.0, but you have importlib-metadata 4.2.0 which is incompatible.
```

**Comment by DavidMStraub** [comment]
Ok. The version pin was added because of this issue https://github.com/python/importlib_metadata/issues/411 which caused builds to fail on 3.7. Can you please try what happens when you remove the version pin in Web API?

**Comment by Samurai336** [comment]
I don't get a dependency conflict when I remove the pin from the setup.py.  pip says it now has 4.2.0 of importlib-metadata installed

**Comment by DavidMStraub** [comment]
Cool, perhaps the bug has been fixed. Do you want to submit a PR?

**Comment by Samurai336** [comment]
I am still having precommit issues. Trying to resolve that is how I ran into this. :)

**Comment by Samurai336** [comment]
You'll have to check me because I just did --no-verify

**Comment by DavidMStraub** [comment]
Fixed by #422.
